sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py

```diff
@@ -324,10 +333,48 @@ def add_collapsable_schema(self):
 
         schema_rest = self._convert_json_schema_to_rest(schema)
         source_name = self.get_sourcename()
-
         for line in schema_rest:
             self.add_line(line, source_name)
 
+    def _get_pydantic_class(self):
+        components_source_name = self.fullname.split('.')
+        mod = __import__(components_source_name[0])
+        for comp in components_source_name[1:]:
+            mod = getattr(mod, comp)
+        return mod
+
+    def add_erdantic_figure(self, collapsed: bool):
+        """Adds an erdantic entity relation diagram to the doc of an pydantic model.
+
+        """
+        source_name = self.get_sourcename()
+        if erd is None:
+            logger = sphinx.util.logging.getLogger(__name__)
+            error_msg = f'erdantic is not installed, no Entity Relationship Diagram for {self.fullname} is skipped.' + \
+                        ' See https://autodoc-pydantic.readthedocs.io/en/stable/users/installation.html'
+            logger.warning(error_msg, location="autodoc_pydantic")
+            self.add_line("..warning::")
+            self.add_line('   ' + error_msg)
+            return
+
+        PydanticClass = self._get_pydantic_class()
```

Comment:
Sphinx' autodoc-directives already hold a reference for the object to be documented via `self.object`. Please use `self.object` and drop `_get_pydantic_class()` completely.